airflow/api/common/mark_tasks.py

```diff
@@ -464,13 +463,16 @@ def set_dag_run_state_to_failed(
 
     # Mark only RUNNING task instances.
     task_ids = [task.task_id for task in dag.tasks]
-    tis = session.query(TaskInstance).filter(
-        TaskInstance.dag_id == dag.dag_id,
-        TaskInstance.run_id == run_id,
-        TaskInstance.task_id.in_(task_ids),
-        TaskInstance.state.in_([State.RUNNING, State.DEFERRED, State.UP_FOR_RESCHEDULE]),
+    tis = session.execute(
+        select(TaskInstance).where(
+            TaskInstance.dag_id == dag.dag_id,
+            TaskInstance.run_id == run_id,
+            TaskInstance.task_id.in_(task_ids),
+            TaskInstance.state.in_([State.RUNNING, State.DEFERRED, State.UP_FOR_RESCHEDULE]),
+        )
     )
-    task_ids_of_running_tis = [task_instance.task_id for task_instance in tis]
+
+    task_ids_of_running_tis = [task_instance.task_id for task_instance, in tis]
```

Comment:
Note: This can be rewritten with `session.scalars` on `TaskInstance.task_id` instead. (Maybe in a later PR.)